### PR TITLE
feat: function to return content of completion script

### PR DESCRIPTION
**Pull request by KSXGitHub**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,10 +7,9 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        node: 
-          - '10'
-          - '12'
-          - '14'
+        node:
+          - '20'
+          - '21'
```
Comment (zkochan) [inline .github/workflows/ci.yml:12]: if it will be shipped to pnpm v9
at least v18 should be supported

File: lib/index.js [hunk 2]
```diff
@@ -7,6 +7,21 @@ const { tabtabDebug, systemShell } = require('./utils');
...
+ * @param {String} options.name - The package configured for completion
+ * @param {String} options.completer - The program the will act as the completer for the `name` program
+ * @param {String} options.shell
+ */
+const print = async ({ name, completer, shell }) => {
+  if (!name) throw new TypeError('options.name is required');
+  if (!completer) throw new TypeError('options.completer is required');
+  if (!shell) throw new TypeError('options.shell is required');
+  const completionScriptContent = await installer.getCompletionScript({ name, completer, shell });
+  console.log(completionScriptContent)
```
Comment (zkochan) [inline lib/index.js:22]: it would make more sense to return the output that pnpm would print out.

**Review by zkochan — CHANGES_REQUESTED** [review]

**Comment by zkochan** [comment]
I don't get it. The completion script is written to the fs, then read and written to the output? Why is it not generated in memory instead and returned from memory?

**Comment by zkochan** [comment]
> The completion script must be exist in the filesystem as a file for the shell programs to load. 

But the referenced issue is exactly about not writing to the fs anything.

**Comment by KSXGitHub** [comment]
> But the referenced issue is exactly about not writing to the fs anything.

It is about **pnpm** not writing to fs for the users. The users themselves must write to fs.

**Comment by zkochan** [comment]
I misunderstood the `readFile` line. It reads the template file that is shipped with pnpm.

**Comment by KSXGitHub** [comment]
> I misunderstood the `readFile` line. It reads the template file that is shipped with pnpm.

It was a terrible name. Fortunately, I wasn't deceived (somehow).

**Pull request merged**